File: SortingAndSearching/BinarySearch.java
import java.util.Arrays;
import java.util.Scanner;

public class Binary_search {
    public static void main(String[] args) {
      
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int[] arr = new int[N];
        for(int i=0;i<N;i++){
            arr[i]=i;
        }
        System.out.println(Arrays.toString(arr));

        System.out.println("Enter a key");
        int key = sc.nextInt();

        int low = 0;
        int high = N-1;
        int flag=0;

        while(low<=high){
            int mid=(low+high)/2;
            if (arr[mid]==key) {
                flag=1;
                break;
            }else{
                if(arr[mid]<key){
                    low=mid+1;
                }else{
                    high=mid-1;
                }
            }
        }

        if(flag==1){
            System.out.println("Key is found");
        }

    }
}

File: SortingAndSearching/BinarySearch.c
#include <stdio.h>
#include <stdlib.h>
 
void BinarySearch(int arr[], int key, int first, int last){
 
   if(first > last)
   {
        printf("Number is not found");
 
   }
	else	{
 
        int mid;
 
        
        mid = (first + last)/2;
 
        
        if(arr[mid]==key){
 
            printf("Element is found at index %d ",mid);
            exit(0);
 
        }else if(arr[mid]>key){
 
            
            BinarySearch(arr, key, first, mid-1);
 
        }else{
 
            
            BinarySearch(arr, key, mid+1, last);
        }
 
    }
 
}
 
int main()
{
	int size;
	printf("Enter the Size:");
	scanf("%d",&size);
	int arr[size];
	printf("Enter the Array:");
	int i;
	for(i=0;i<size;i++)
	{
		scanf("%d",&arr[i]);
	}
 
    
   
   int key;
   	printf("Enter the number to be search:");
	scanf("%d",&key);
 
    int first =0 , last = (sizeof(arr)/sizeof(arr[0]))-1;
 
   BinarySearch(arr, key, first, last);
}
